main.go

```diff
@@ -71,6 +71,7 @@ var (
 	serverUnsafeTarget   = serverCommand.Flag("unsafe-target", "If set, does not limit target to localhost, 127.0.0.1, [::1], or UNIX sockets.").Bool()
 	serverAllowAll       = serverCommand.Flag("allow-all", "Allow all clients, do not check client cert subject.").Bool()
 	serverAllowedCNs     = serverCommand.Flag("allow-cn", "Allow clients with given common name (can be repeated).").PlaceHolder("CN").Strings()
+	serverAllowedUIDs    = serverCommand.Flag("allow-uid", "Allow clients with given uid (can be repeated).").PlaceHolder("UID").Strings()
```

Comment:
Nit: UID should be upper-case here.

Reply:
You're also missing the equivalent flag for client(s)/client mode.

Reply:
I assume you're referencing UID within the description?

I couldn't think of a scenario where a a secure endpoint would present a certificate with the UID field. This option seems very user/automation specific. I am happy to add it, though.

Reply:
I'd prefer to have it symmetric since it's not much to add. Also since you already implemented it in the auth code.